### PR TITLE
Fix issue #642, frontend not in english if the user language does not exist

### PR DESCRIPTION
**Pull request by requiem958**
Fix [#642](https://github.com/Plume-org/Plume/issues/642)

Iterate over catalogs to find the english index and then use it instead of 0.

### CODE REVIEW

File: plume-front/src/main.rs [hunk 1]
```diff
@@ -46,10 +46,15 @@ lazy_static! {
         let catalogs = include_i18n!();
         let lang = js! { return navigator.language }.into_string().unwrap();
         let lang = lang.splitn(2, '-').next().unwrap_or("en");
+
+        let english_position = catalogs
```
Comment (igalic) [inline plume-front/src/main.rs]: this should be either a (cached?) constant, or a (cached?) config value

File: plume-front/src/main.rs [hunk 2]
```diff
@@ -46,10 +46,15 @@ lazy_static! {
...
+            .iter()
+            .position(|(language_code, _)| *language_code == "en")
+            .unwrap();
```
Comment (igalic) [inline plume-front/src/main.rs]: while i understand that this is a simple operation, i don't think we need to do it every time

Comment (igalic) [reply plume-front/src/main.rs]: actually, never mind, this is an operation that's only dealing with things known at compile time

the optimiser will probably inline this as a constant literal

File: plume-front/src/main.rs (hunk 1 above)
Comment (requiem958) [reply plume-front/src/main.rs]: It should be a constant yes, but the rust doc of constant says : 

> Constants can be set only to a constant expression and not to the result of a function call or any other value that will be computed at runtime.
So i thought the keyword const was forbidden in this case.

Comment (fdb-hiroshima) [reply plume-front/src/main.rs]: All this is in a lazy_static! call, so the result will be evaluated only one time, the first time CATALOG is accessed. The only thing it's not is cached between two different page loading, but that's not necessary imo

**Pull request merged**